core/algorithms/onpolicy_sync/engine.py

```diff
@@ -146,18 +146,20 @@ def __init__(
 
         self._vector_tasks: Optional[VectorSampledTasks] = None
 
-        self.observation_set = None
+        self.sensor_preprocessor_graph = None
```

Comment:
Hindsight is 20/20 :)

Really I would like to somehow incorporate the preprocessor graph into the `SensorSuite` class. This would have the very pleasing result of keeping everything regarding sensors in a single place. The problem of course is that the `SensorSuite` is used at the task sampler level while the preprocessors often need to be called after accumulating all the observations. We could try some tricks to handle this but it becomes a bit onerous.

Reply:
I think it's a good idea. If the SensorSuite knows which Sensors in the Suite are directly applicable for one instance, then we only apply those during task sampling, then the rest are applied after data collection. I already do something similar with the visualizer, where the same class is used in two different ways by the engine and the runner/logger...